### PR TITLE
operator/config: Introduce polymorphic advertised API config

### PR DESCRIPTION
**Pull request by RafalKorepta**
The config in rpk was changed to support polymorphic advertised
kafka api. The latest redpanda container does not incorporate
mentioned change. To not create braking change for the user
operator introduces feature flag that will enable polymorphic
advertised kafka api in the configuration. The feature flag need
to be used after version v21.2.1 of the redpanda container.

## Checklist
- [ ] Reference related [issue](https://github.com/vectorizedio/redpanda/issues)
- [ ] Update [PendingReleaseNotes.md](https://github.com/dotnwat/redpanda/blob/dev/PendingReleaseNotes.md), if relevant

When referencing a related issue, remember to migrate duplicate stories from the
external tracker. This is not relevant for most users.

### CODE REVIEW

File: src/go/k8s/pkg/resources/configmap.go [hunk 1]
```diff
@@ -145,49 +149,83 @@ func (r *ConfigMapResource) Obj() (k8sclient.Object, error) {
...
-		RPCServer: config.SocketAddress{
-			Address:	"0.0.0.0",
-			Port:		rpcServerPort,
-		},
-		AdvertisedRPCAPI:	&config.SocketAddress{},
-		KafkaApi: []config.NamedSocketAddress{
-			{
-				SocketAddress: config.SocketAddress{
+		return &Config{
+			ConfigFile:	"/etc/redpanda/redpanda.yaml",
```
Comment (dimitriscruz) [inline src/go/k8s/pkg/resources/configmap.go]: Please replace `/etc/redpanda/redpanda.yaml` with `configPath`, it's set at the top.

**Review by dimitriscruz — CHANGES_REQUESTED** [review]
Please add to the commit message something like the following: 
- When the feature flag is off, rpk's old config struct (copied here) is used to generate the configuration file, otherwise we base the configuration on rpk's new default that supports multiple socket addresses.
- Once the `latest` image is updated with this change, we need to switch the default value of the flag, or remove it and the corresponding path entirely.

File: src/go/k8s/pkg/resources/configmap.go (hunk 1 above)
Comment (RafalKorepta) [reply src/go/k8s/pkg/resources/configmap.go]: Thanks. Will do

**Comment by RafalKorepta** [comment]
@dimitriscruz commit message changed.

**Review by dimitriscruz — APPROVED** [review]

**Pull request merged**